ast-nodes/expression.dart

```diff
@@ -1,16 +1,85 @@
+import 'boolean-literal.dart';
+import 'index.dart';
 import 'statement.dart';
 import 'and-operator.dart';
 import '../lexer.dart';
+import '../iterator-utils.dart';
+import '../syntax-error.dart';
 
 /// An abstract expression that returns a value.
 abstract class Expression implements Statement {
   factory Expression.parse(Iterable<Token> tokens) {
-    // TODO: write the actual parser body
-    return AndOperator(null, null);
+    bool notFlag = false;
+    String operand;
+    var iterator = tokens.iterator;
+
+    if (tokens.isEmpty) {
+      return AndOperator(BooleanLiteral(!notFlag), Comparison.parse([]));
+    }
+    notFlag = tokens.first.value == "not";
+    if (notFlag) {
+      checkNext(iterator, RegExp("\^not\$"), "internal error");
```

Comment:
Don't backslash-escape the "^", only the dollar needs escaping, this might even cause an error.
And what is this check supposed to check? You've already ensured that the first token is a `not`, why do you expect the second token to also be a `not`?